File: Frontend/YourPubQuiz/src/app/core/models/quizResult.ts
export interface QuizResult {
  totalQuestions: number;
  correctAnswers: number;
  questionResults: QuestionResult[];
}

export interface QuestionResult{
  questionId: string;
  questionText: string;
  isCorrect: boolean;
  userAnswer: string;
  correctAnswer: string;
}

File: YourPubQuiz/Models/QuizResult.cs
namespace YourPubQuiz.Models;

public class QuizResult
{
    public int TotalQuestions { get; set; }
    public int CorrectAnswers { get; set; }
    public List<QuestionResultDetails> QuestionResults { get; set; } = [];
}

public class QuestionResultDetails
{
    public string QuestionId { get; set; }
    public string QuestionText { get; set; }
    public bool IsCorrect { get; set; }
    public string UserAnswer { get; set; }
    public string CorrectAnswer { get; set; }
}